File: Back/Model/Product/Product.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
namespace Model
{
    public class Product
    {
        [Key]
        public int ProductID { get; set; }
        public string Name { get; set; }
        public bool Enabled{ get; set; }
        public Nation Nation { get; set; }
        public Brand Brand { get; set; }
        public List<ProductScore> ProductScore { get; set; }
        public List<ProductComment> ProductComment { get; set; }
        public List<ProductMultiMedia> ProductMultiMedia { get; set; }
        public List<ProductSeller> ProductSeller { get; set; }
        public List<ProductSimilar> ProductSimilar { get; set; }
    }
}

File: Front/src/app/Model/Product/Product.ts
import { Nation } from './Nation';
import { Brand } from './Brand';
import { ProductScore } from './ProductScore';
import { ProductComment } from './ProductComment';
import { ProductMultiMedia } from './ProductMultiMedia';
import { ProductSeller } from './ProductSeller';
import { ProductSimilar } from './ProductSimilar';

export interface Product {
    productID: number;
    name: string;
    enabled: boolean;
    nation: Nation;
    brand: Brand;
    productScore: ProductScore[];
    productComment: ProductComment[];
    productMultiMedia: ProductMultiMedia[];
    productSeller: ProductSeller[];
    productSimilar: ProductSimilar[];
}
